app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt

```diff
@@ -76,10 +76,12 @@ class CustomDrugEntryUpdate : Update<CustomDrugEntryModel, CustomDrugEntryEvent,
       patientUuid: UUID,
       dosagePlaceHolder: String
   ): Next<CustomDrugEntryModel, CustomDrugEntryEffect> {
+    val dosage = if (model.dosage == dosagePlaceHolder) null else model.dosage
```

Comment:
Instead of passing the placeholder, can't we just check if the dosage contains any numeric value or not? WDYT?

Reply:
uumm yeah 👍🏻

Reply:
@msasikanth 
Does using regex to handle this sound like a good approach?

  ```
  val atleastOneDigitRegex = "\\d".toRegex()
    val dosage = if (atleastOneDigitRegex.containsMatchIn(model.dosage!!)) model.dosage else null

```

Reply:
Yeah, that works. But we can probably go with this approach instead?

```kt
val isAValidDosage = dosage.filter { it.isDigit() }.isNotBlank()
```